lib/src/legacy/utils.ts

```diff
@@ -38,7 +38,13 @@ export function removeLegacyImporter(string: string): string {
 // syntax.
 export function removeLegacyImporterFromSpan(span: SourceSpan): SourceSpan {
   if (!span.url) return span;
-  return {...span, url: new URL(removeLegacyImporter(span.url.toString()))};
+  const url = removeLegacyImporter(span.url.toString());
+  try {
+    return {...span, url: new URL(url)};
+  } catch (_) {
```

Comment:
If the URL is absolute, it won't use the basename anyway, so you can just pass it in eagerly rather than catching this error.